tests/validation/cattlevalidationtest/core/test_cluster.py

```diff
@@ -0,0 +1,119 @@
+        # push out socat so swarm works
+        socat_container1 = client.create_container(
+            name=test_name + '-socat',
+            networkIds=[managed_network.id],
+            imageUuid=socat_test_image,
+            ports='2375:2375/tcp',
+            stdinOpen=False,
+            tty=False,
+            dataVolumes='/var/run/docker.sock:/var/run/docker.sock',
+            requestedHostId=hosts[0].id)
+        wait_for_condition(
+            client, socat_container1,
+            lambda x: x.state == 'running',
+            lambda x: 'State is: ' + x.state)
+
+        socat_container2 = client.create_container(
+            name=test_name + '-socat',
+            networkIds=[managed_network.id],
+            imageUuid=socat_test_image,
+            ports='2375:2375/tcp',
+            stdinOpen=False,
+            tty=False,
+            dataVolumes='/var/run/docker.sock:/var/run/docker.sock',
+            requestedHostId=hosts[1].id)
+        wait_for_condition(
+            client, socat_container2,
+            lambda x: x.state == 'running',
+            lambda x: 'State is: ' + x.state)
+
+        cluster = client.create_cluster(name=test_name + '-cluster',
+                                        port=9000)
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: x.state == 'inactive',
+            lambda x: 'State is: ' + x.state)
+
+        cluster.addhost(hostId=hosts[0].id)
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: len(x.hosts()) == 1,
+            lambda x: 'Number of hosts in cluster is: ' + len(x.hosts()))
+
+        cluster.activate()
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: x.state == 'active',
+            lambda x: 'State is: ' + x.state)
+
+        # check Cluster Server is deployed
+        cluster_server = None
+        for instance in cluster.instances():
+            if (instance.state == 'removed'):
+                continue
+            if (instance.name == 'Cluster Server'):
+                cluster_server = instance
+                break
+
+        assert cluster_server is not None
+
+        assert cluster_server.state == 'running'
+
+        cluster.addhost(hostId=hosts[1].id)
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: len(x.hosts()) == 2,
+            lambda x: 'Number of hosts in cluster is: ' + len(x.hosts()))
+
+        # deploy container to cluster
+        test_container = client.create_container(
+            name=test_name + "-cl-deployed",
+            networkIds=[managed_network.id],
+            imageUuid='docker:ubuntu',
+            requestedHostId=cluster.id)
+        wait_for_condition(
+            client, test_container,
+            lambda x: x.state == 'running',
+            lambda x: 'State is: ' + x.state)
+
+        test_container_found = None
+        for instance in cluster.instances():
```

Comment:
Sorry.  Accidentally closed the PR.
I updated the test to try to make more checks regarding the network.  But I wasn't able to find the vnet or subnet in the API.
cluster.instances() I think makes an API call so it doesn't look like cluster itself needs to be updated unless I'm checking fields in that specific resource.